tests/Functional/EventListener/SwitchUserListenerTest.php

```diff
@@ -28,12 +32,18 @@
 class SwitchUserListenerTest extends WebTestCase
 {
     use MockeryPHPUnitIntegration;
+    use SessionHelperTrait;
+
+    private const FIREWALL_NAME = 'secured_area';
+    private const SESSION_ID = 'test';
+    private $sessionName;
+    private static $overrideService = false;
 
     public function testSwitchUserCompatibility()
     {
         $client = static::createClient();
-        $session = $client->getContainer()->get('session');
-        $this->loginAsAdmin($client, $session);
+        $this->callInRequestContext($client, [$this, 'loginAsAdmin']);
```

Comment:
@wouterj inspired by https://github.com/symfony/symfony/pull/41883 i tried this, but it seems to not work as expected, neither for symfony 5 not symfony 6.

do you see what i am doing wrong, by any chance?

Reply:
@acrobat do you by any chance see what i do wrong with the session manipulation in the refactoring?

Reply:
@dbu I will take a look at it!